mapper.go

```diff
@@ -0,0 +1,308 @@
+}
+
+// MapParser is a parser to map.
+// Only `json` is supported now.
+type MapParser struct{}
+
+// NewMapParser creates MapParser.
+func NewMapParser() *MapParser {
+	return &MapParser{}
+}
+
+// Map is the implementation of Mapper[string] for MapParser.
+func (*MapParser) Map(src string) (any, error) {
+	var result any
+	err := json.Unmarshal([]byte(src), &result)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// SliceParser is a parser to slice.
+// Only `json` is supported now.
+type SliceParser struct{}
+
+// NewSliceParser creates SliceParser.
+func NewSliceParser() *SliceParser {
+	return &SliceParser{}
+}
+
+// Map is the implementation of Mapper[string] for SliceParser.
+func (*SliceParser) Map(src string) (any, error) {
+	var result any
+	err := json.Unmarshal([]byte(src), &result)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// BinaryParser is a parser to binary.
+type BinaryParser struct{}
+
+// NewBinaryParser creates BinaryParser.
+func NewBinaryParser() *BinaryParser {
+	return &BinaryParser{}
+}
+
+// Map is the implementation of Mapper[string] for BinaryParser.
+func (*BinaryParser) Map(src string) (any, error) {
+	return []byte(src), nil
+}
+
+// NullParser is a parser to nil.
+type NullParser struct {
+	nullValue string
+}
+
+// NewNullParser creates NullParser.
+func NewNullParser(nullValue string) *NullParser {
+	return &NullParser{nullValue: nullValue}
+}
+
+// Map is the implementation of Mapper[string] for NullParser.
+func (parser *NullParser) Map(src string) (any, error) {
+	if src == parser.nullValue {
+		return nil, nil
+	}
+	return nil, NewErrUnexpectedValue(src)
+}
+
+// IntervalParser is a parser to datetime.Interval.
+type IntervalParser struct{}
+
+// NewIntervalParser creates IntervalParser.
+func NewIntervalParser() *IntervalParser {
+	return &IntervalParser{}
+}
+
+// intervalFieldsNumber is the number of fields in datetime.Interval.
+const intervalFieldsNumber = 9
+
+// Map is the implementation of Mapper[string] for IntervalParser.
+func (*IntervalParser) Map(src string) (any, error) {
+	parts := strings.Split(src, ",")
+	if len(parts) != intervalFieldsNumber {
+		return nil, NewErrUnexpectedValue(src)
+	}
+	partsAsInt64 := [intervalFieldsNumber]int64{}
+	for i, part := range parts {
+		var err error
+		if partsAsInt64[i], err = strconv.ParseInt(part, 10, 64); err != nil {
+			return nil, NewErrUnexpectedValue(src)
+		}
+	}
+	adjust := datetime.Adjust(partsAsInt64[8])
+	if adjust != datetime.NoneAdjust &&
+		adjust != datetime.ExcessAdjust && adjust != datetime.LastAdjust {
+		return nil, NewErrUnexpectedValue(src)
+	}
+	interval := datetime.Interval{
+		Year:   partsAsInt64[0],
+		Month:  partsAsInt64[1],
+		Week:   partsAsInt64[2],
+		Day:    partsAsInt64[3],
+		Hour:   partsAsInt64[4],
+		Min:    partsAsInt64[5],
+		Sec:    partsAsInt64[6],
+		Nsec:   partsAsInt64[7],
+		Adjust: adjust,
+	}
+	return interval, nil
+}
```

Comment:
Do you have an example of a string that describes an interval?

```
> tostring(require('datetime').interval.new{month = 1, adjust = 'last'})
---
- +1 months
```

Reply:
There are some examples in tests. 
It was decided to support only this format initially and complete it upon request.

Reply:
Where is this format could be found in practice? What is application exports an interval to this format?

Reply:
tt export is supposed to use this format. 
(For now, export doesn't support intervals).

Reply:
> tt export is supposed to use this format.

Where is this requirement described?

Reply:
So, does interval type support parsing "+1 months" format back to interval?

Reply:
> If no, this should be done in `go-tarantool` under a separate issue, not in `tupleconv`.

I do not agree with such logic. The purpose of the external types in go-tarantool is to provide types with the ability to import/export to msgpack. Importing and exporting string representations of types is the business logic of an application. We see it clearly in this discussion.

I don't mind if you import/export to your own format `1,2,3,4,...2`, but in practice it would be confusing when there is a human-readable format already exists.

Reply:
So, if I ever want to work with some encapsulated type (no public fields) I will not have an option to implement formatter and parser in a separate package. And we will have formatters and parsers in two different repositories: some types will have them nearby, other types - in `tupleconv`. That is why making decimal from string is implemented in `go-tarantool`, because an external code has no access to the internal `decimal` field.

Reply:
> Importing and exporting string representations of types is the business logic of an application.

If a business logic wants some specific formatting, the type must provide a way to do so. Creating a struct type with all fields external and shifting formatting/parsing responsibility to an application is not good. Usually the types provide a way to print using default format and some formatting modes if required. For `go-tarantool` types default format may be some kind of "tarantool style format".

Reply:
> That is why making `Decimal` from string is implemented in `go-tarantool`

It's hard for the user to create a `Decimal` value not from a string.

> because an external code has no access to the internal `decimal` field.

An external code has access to the Go-representation of the `Decimal` type.

```Go
	dec := MakeDecimal(decimal.Decimal{})
	fmt.Println(dec.Decimal)
	fmt.Println(dec.Sign())
	// Output:
	// 0
	// 0
```

Reply:
> It's hard for the user to create a Decimal value not from a string.

Not so hard. I can create it simply from float or integers for example. Why is interval different?

> No, it is a logic of an application, not a type.

It may not type logic, but definitely package responsibility. If I read package documentation and see some value type like `interval` I expect to find printing function nearby. I don't want to look for another package just to print it in human-readable form.